cdk/cdk/cdk_stack.py

```diff
@@ -58,13 +76,26 @@ def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
             self, "Generate COG Task",
             lambda_function=generate_cog_lambda
         )
+        generate_stac_item_task = tasks.LambdaInvoke(
+            self, "Generate STAC Item Task",
+            lambda_function=generate_stac_item
+        )
+
         map_cogs = stepfunctions.Map(self, "Map State",
             max_concurrency=10,
             items_path=stepfunctions.JsonPath.string_at("$.Payload")
         )
         map_cogs.iterator(generate_cog_task)
 
-        definition = start_state.next(discover_task).next(map_cogs)
+        map_stac_items = stepfunctions.Map(self, "Map State",
+            max_concurrency=10,
+            items_path=stepfunctions.JsonPath.string_at("$.Payload")
+        )
+        map_stac_items.iterator(generate_stac_item_task)
+
+        processing_tasks = stepFunctions.Parallel(self, 'All jobs').branch(map_cogs).branch(map_stac_items)
```

Comment:
I think we may need to chain the generate_cog and generate_stac tasks so that the stac item generation is executed in sequence (so that it can include the cog asset in the stac record). I haven't used the fan out functionality in step functions before but it seems like we could define that chain of generating cog and stac metadata and then fan out in parallel to that chain?

Or do both the hdf to cog and stac metadata generation in the same step. I know we had mixed feelings about that in the tag-up, though.